src/muse/costs.py

```diff
@@ -276,31 +287,45 @@ def lifetime_levelized_cost_of_energy(
 
     # Fuel/energy costs
     prices_fuel = filter_input(prices, commodity=fuels, year=years.values)
-    fuel = consumption(technologies=techs, production=production, prices=prices)
-    fuel_costs = (fuel * prices_fuel * rates).sum(("commodity", "year"))
+    fuel_costs = (cons * prices_fuel * rates).sum(("commodity", "year"))
 
     # Cost related to material other than fuel/energy and environmentals
     prices_material = filter_input(prices, commodity=material, year=years.values)
-    material_costs = (production * prices_material * rates).sum(("commodity", "year"))
+    material_costs = (cons * prices_material * rates).sum(("commodity", "year"))
 
-    # Fixed and Variable costs
-    fixed_costs = convert_timeslice(
-        techs.fix_par * (capacity**techs.fix_exp),
-        prices.timeslice,
-        QuantityType.EXTENSIVE,
+    # Fixed costs
+    fixed_costs = (
+        convert_timeslice(
+            techs.fix_par * (capacity**techs.fix_exp),
+            prices.timeslice,
+            QuantityType.EXTENSIVE,
+        )
+        * rates
+    ).sum("year")
+
+    # Variable costs
+    tech_activity = (production / techs.fixed_outputs).max("commodity")
+    variable_costs = ((techs.var_par * tech_activity**techs.var_exp) * rates).sum(
+        "year"
     )
-    variable_costs = (
-        techs.var_par * production.sel(commodity=products) ** techs.var_exp
-    ).sum("commodity")
-    fixed_and_variable_costs = ((fixed_costs + variable_costs) * rates).sum("year")
-    denominator = production.where(production > 0.0, 1e-6)
+
+    # Production
+    prod = (
+        production.where(production > 0.0, 1e-6)
```

Comment:
I always thought this `1e-6` was quite arbitrary. Not sure what a better way of dealing with it would be, though.